File: Projects/05/root.clj
;;; Emily Costa
;;; March 11, 2020
;;; CS 214 A
;;; Project 5 -  
;;; Calculate the roots using the quadratic equaiton
;;; given values for a, b, and c from the user Math/sqrt
;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;

(ns root)

(defn calculate[a b c]

	;(let [
		(def disc (- (* b b) (* 4 (* a c))) )
		;(def root1 (float ( / (+ ((- 0.0 (float b)) (float (Math/sqrt disc)))) (* 2.0 (float a) ))))
		(def root1 (float (Math/sqrt disc)))
		;(def root2 (float ( / (- ((- 0.0 b) (float (Math/sqrt disc)))) (* 2.0 a)))) ;]
		;root1 (float (Math/sqrt disc))
		(def root2 (float 3))

		(vector disc root1 root2)
	
	;(if (< disc 0)

		;(vector 0 0 0))

	;(if (= disc 0)

		;(vector 1 root1 0))

	;( if (> disc 0)

		;(vector 2 root1 root2))

	;) ; end let

) ; end calculate function

(defn -main[]

	(printf "\n To calculate the root(s): ")
	(print "\n Enter a value for a, b, and c: ") (flush)
	(let [ 
		a (float(read)) 
		b (float(read))
		c (float(read)) 

		result (calculate a b c)
		part1 (get result 0)
		part2 (get result 1)
		part3 (get result 2)
		]

	(print part1)
	(printf "\n")
	(print part2)
	(printf "\n")


	;(printf " disc is: %.2f b is: %.2f, c is: %.2f \n\n" (float part1) (float part2) (float part3))

	
	) ; end let
) ; end main

File: Projects/05/root.java
/*
* Emily Costa
* March 11, 2020
* CS 214 A
* Project 5 -  
* Calculate the roots using the quadratic equaiton
* given values for a, b, and c from the user
*/

import java.util.Scanner;

public class root {

	public static void calculate (double a, double b, double c, double[] results) {

		double disc = Math.pow(b,2) - (4 * a * c);

		if (disc < 0){
			results[0] = 0;
			return;
		}
		else if (disc == 0) {
			results[0] = 1;
			results[1] = (-b + Math.sqrt(disc)) / (2 * a);
			return; 
		}
		else if (disc > 0) {
			results[0] = 2;
			results[1] = (-b + Math.sqrt(disc)) / (2 * a);
			results[2] = (-b - Math.sqrt(disc)) / (2 * a);
			return;
		}
		

	} // end of method calculate

	public static void main (String[] args) {

		// promt user and scan in string
		System.out.print("\n To calculate the root(s): ");
		System.out.print("\n Enter a value for a, b, and c: ");
		// create a connection names keyboard
		Scanner keyboard = new Scanner(System.in);
		// read in values
		double a = keyboard.nextDouble();
		double b = keyboard.nextDouble();
		double c = keyboard.nextDouble();

		// create an array to hold the roots
		double[] resultList = new double[3];
		// call function to create the substrings
		calculate(a,b,c,resultList);
		// if statements to print the proper results
		if (resultList[0] == 0){
			System.out.print(" There are no real solutions. \n\n");
		}
		else if (resultList[0] == 1){
			System.out.printf(" The root is %.2f \n\n", resultList[1]);
		}
		else if (resultList[0] == 2){
			System.out.printf(" The roots are %.2f and %.2f \n\n", resultList[1], resultList[2]);
		}

	} // end of main method

} // end class root